packages/dds/tree/src/test/feature-libraries/editable-tree/editableTree.edit.spec.ts

```diff
@@ -0,0 +1,329 @@
+    localFields: {
+        [EmptyKey]: fieldSchema(FieldKinds.sequence, [stringSchema.name]),
+    },
+    extraLocalFields: emptyField,
+});
+
+const addressSchema = namedTreeSchema({
+    name: brand("Test:Address-1.0.0"),
+    localFields: {
+        street: fieldSchema(FieldKinds.value, [stringSchema.name]),
+        zip: fieldSchema(FieldKinds.optional, [stringSchema.name]),
+        phones: fieldSchema(FieldKinds.optional, [phonesSchema.name]),
+        simplePhones: fieldSchema(FieldKinds.optional, [simplePhonesSchema.name]),
+        sequencePhones: fieldSchema(FieldKinds.sequence, [stringSchema.name]),
+    },
+    extraLocalFields: emptyField,
+});
+
+const mapStringSchema = namedTreeSchema({
+    name: brand("Map<String>"),
+    extraLocalFields: fieldSchema(FieldKinds.value, [stringSchema.name]),
+});
+
+const personSchema = namedTreeSchema({
+    name: brand("Test:Person-1.0.0"),
+    localFields: {
+        name: fieldSchema(FieldKinds.value, [stringSchema.name]),
+        age: fieldSchema(FieldKinds.value, [int32Schema.name]),
+        salary: fieldSchema(FieldKinds.value, [float32Schema.name]),
+        friends: fieldSchema(FieldKinds.value, [mapStringSchema.name]),
+        address: fieldSchema(FieldKinds.value, [addressSchema.name]),
+    },
+    extraLocalFields: emptyField,
+});
+
+const optionalChildSchema = namedTreeSchema({
+    name: brand("Test:OptionalChild-1.0.0"),
+    localFields: {
+        child: fieldSchema(FieldKinds.optional),
+    },
+    value: ValueSchema.Serializable,
+    extraLocalFields: emptyField,
+});
+
+const schemaTypes: Set<NamedTreeSchema> = new Set([
+    optionalChildSchema,
+    stringSchema,
+    float32Schema,
+    int32Schema,
+    complexPhoneSchema,
+    phonesSchema,
+    simplePhonesSchema,
+    addressSchema,
+    mapStringSchema,
+    personSchema,
+]);
+
+const schemaMap: Map<TreeSchemaIdentifier, NamedTreeSchema> = new Map();
+for (const named of schemaTypes) {
+    schemaMap.set(named.name, named);
+}
+
+const rootPersonSchema = fieldSchema(FieldKinds.value, [personSchema.name]);
+
+const fullSchemaData: SchemaData = {
+    treeSchema: schemaMap,
+    globalFieldSchema: new Map([[rootFieldKey, rootPersonSchema]]),
+};
+
+// TODO: derive types like these from those schema, which subset EditableTree
+
+type Int32 = Brand<number, "Int32">;
+const newAge: Int32 = brand(55);
+
+type ComplexPhoneType = EditableTree & {
+    number: string;
+    prefix: string;
+};
+
+type PhonesType = UnwrappedEditableSequence & (number | string | ComplexPhoneType)[];
+
+type SimplePhonesType = UnwrappedEditableSequence & string[];
+
+type AddressType = EditableTree & {
+    street: string;
+    zip?: string;
+    phones?: PhonesType;
+    simplePhones?: SimplePhonesType;
+    sequencePhones?: SimplePhonesType;
+};
+
+type PersonType = EditableTree & {
+    name: string;
+    age: Int32;
+    salary: number;
+    friends: Record<string, string>;
+    address: AddressType;
+};
```

Comment:
A lot of this seems duplicated from `editableTree.spec.ts`. Please consider pulling it into a separate file so it can be de-duplicated.

Reply:
done